workflows/cloud_run_job/main.tf

```diff
@@ -0,0 +1,139 @@
+  location = "us-central1"
+
+  template {
+    task_count = 10
+    template {
+      containers {
+        image   = "us-central1-docker.pkg.dev/PROJECT_ID/REPOSITORY/parallel-job:latest"
+        command = ["python"]
+        args    = ["process.py"]
+        env {
+          name  = "INPUT_BUCKET"
+          value = "input-PROJECT_ID"
+        }
+        env {
+          name  = "INPUT_FILE"
+          value = "input_file.txt"
+        }
+      }
+    }
+  }
+}
+# [END workflows_terraform_create_run_job]
+
+# [START workflows_terraform_create_workflow]
+# Create a workflow
+resource "google_workflows_workflow" "default" {
+  name        = "cloud-run-job-workflow"
+  region      = "us-central1"
+  description = "Workflow that routes a Cloud Storage event and executes a Cloud Run job"
+
+  # Note that $$ is needed for Terraform
+  source_contents = <<EOF
+  main:
+      params: [event]
+      steps:
+          - init:
+              assign:
+                  - project_id: $${sys.get_env("GOOGLE_CLOUD_PROJECT_ID")}
+                  - event_bucket: $${event.data.bucket}
+                  - event_file: $${event.data.name}
+                  - target_bucket: $${"input-" + project_id}
+                  - job_name: parallel-job
+                  - job_location: us-central1
+          - check_input_file:
+              switch:
+                  - condition: $${event_bucket == target_bucket}
+                    next: run_job
+                  - condition: true
+                    next: end
+          - run_job:
+              call: googleapis.run.v1.namespaces.jobs.run
+              args:
+                  name: $${"namespaces/" + project_id + "/jobs/" + job_name}
+                  location: $${job_location}
+                  body:
+                      overrides:
+                          containerOverrides:
+                              env:
+                                  - name: INPUT_BUCKET
+                                    value: $${event_bucket}
+                                  - name: INPUT_FILE
+                                    value: $${event_file}
+              result: job_execution
+          - finish:
+              return: $${job_execution}
+  EOF
+}
+# [END workflows_terraform_create_workflow]
+
+# [START workflows_terraform_create_trigger]
+# Create an Eventarc trigger that routes Cloud Storage events to Workflows
+resource "google_eventarc_trigger" "default" {
+  name     = "cloud-run-job-trigger"
+  location = google_workflows_workflow.default.region
+
+  # Capture objects changed in the bucket
+  matching_criteria {
+    attribute = "type"
+    value     = "google.cloud.storage.object.v1.finalized"
+  }
+  matching_criteria {
+    attribute = "bucket"
+    value     = google_storage_bucket.default.name
+  }
+
+  # Send events to Workflows
+  destination {
+    workflow = google_workflows_workflow.default.id
+  }
+
+  service_account = "SERVICE_ACCOUNT_NAME@PROJECT_ID.iam.gserviceaccount.com"
```

Comment:
Could you add a service account resource and reference it here like in this sample so: https://github.com/terraform-google-modules/terraform-docs-samples/blob/3e9f924366befdeb26d1046a6479ed1e9c4d8bb6/run/secret_manager/main.tf#L57

Reply:
Done